File: CS50X 2023/Pset6/sentimental-mario-less/mario.py
# Maimoona Aziz
# TODO


def main():
    # Prompt for height more than one and less than 8
    height = 0
    while True:
        try:
            height = int(input("Height: "))
            if 1 <= height <= 8:
                break
        except ValueError:
            print("Invalid input")

    # print_walls
    print_walls(height)


# function for printing walls
def print_walls(height):
    for high in range(1, height + 1):  # For height
        for space in range(1, height - high + 1):
            print(" ", end="")  # Printing spaces
        for wide in range(1, high + 1):  # For width
            print("#", end="")  # Printing hashes
        print()


main()

File: CS50X 2023/Pset1/mario-less/mario.c
// Maimoona Aziz
// Mario less confident
#include <cs50.h>
#include <stdio.h>

void print_walls(int height);
int main(void)
{
    // Ask for height ; keep asking if value less than 1 and more than 8
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // Print out # pyramid
    print_walls(height);
}

// Function for building pyramid
void print_walls(int height)
{
    for (int high = 1; high <= height; high++)
    {
        // Print spaces (spaces are total height - current row)
        for (int space = 1; space <= (height - high); space++)
        {
            printf(" ");
        }

        // Print # pyramid
        for (int wide = 1; wide <= high; wide++)
        {

            printf("#");
        }

        // Next line (row)
        printf("\n");
    }
}